ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -0,0 +1,1306 @@
+                is_type = str(get(self._switch_data_combined, "is_type", default=default_is_type)).lower()
+                if is_type not in ["level-1", "level-2", "level-1-2"]:
+                    is_type = "level-2"
+                return is_type
+        return None
+
+    @cached_property
+    def isis_instance_name(self):
+        if self.underlay_router is True:
+            if self.underlay_routing_protocol in ['isis', 'isis-ldp', 'isis-sr', 'isis-sr-ldp']:
+                if self.mpls_lsr is True:
+                    default_isis_instance_name = "CORE"
+                else:
+                    default_isis_instance_name = "EVPN_UNDERLAY"
+                return get(self._hostvars, "underlay_isis_instance_name", default=default_isis_instance_name)
+        return None
+
+    @cached_property
+    def node_sid(self):
+        if self.underlay_router is True:
+            if self.underlay_routing_protocol in ['isis-sr', 'isis-sr-ldp']:
+                node_sid_base = int(get(self._switch_data_combined, "node_sid_base", 0))
+                return self.id + node_sid_base
+        return None
+
+    @cached_property
+    def loopback_ipv6_pool(self):
+        if get(self._hostvars, "underlay_ipv6") is True:
+            return get(self._switch_data_combined, "loopback_ipv6_pool", required=True)
+        return None
+
+    @cached_property
+    def loopback_ipv6_offset(self):
+        if get(self._hostvars, "underlay_ipv6") is True:
+            return get(self._switch_data_combined, "loopback_ipv6_offset", default=0)
+        return None
+
+    @cached_property
+    def ipv6_router_id(self):
+        '''
+        Run template lookup to render ipv6 address for router_id
+
+        Since some templates might contain certain legacy variables (switch_*),
+        those are mapped from the switch.* model
+        '''
+        if get(self._hostvars, "underlay_ipv6") is True:
+            template_vars = {"ansible_search_path": self._ansible_search_path}
+            # Copying __dict__ will expose all switch facts cached up until this function is run.
+            # TODO: We should probably find and document a list of supported context variables instead.
+            template_vars['switch'] = {k: self.__dict__.get(k) for k in self.keys()}
+            template_vars['switch_id'] = self.id
+            template_vars['loopback_ipv6_pool'] = self.loopback_ipv6_pool
+            template_vars['loopback_ipv6_offset'] = self.loopback_ipv6_offset
+            template_path = get(self.ip_addressing, "ipv6_router_id", required=True)
+            return template_var(template_path, template_vars, self._template_lookup_module)
+        return None
+
+    @cached_property
+    def mlag(self):
+        if (
+            self.mlag_support is True
+            and get(self._switch_data_combined, "mlag", default=True) is True
+            and len(self._switch_data_node_group_nodes) == 2
+        ):
+            return True
+        return False
```

Comment:
```suggestion
    def mlag(self):
        return (
            self.mlag_support is True
            and get(self._switch_data_combined, "mlag", default=True) is True
            and len(self._switch_data_node_group_nodes) == 2
        )
```

no need to return True False as we literally return the boolean value if the comparison